render-wasm/src/render/fills.rs

```diff
@@ -56,7 +56,12 @@ fn draw_image_fill_in_container(
     // Set the clipping rectangle to the container bounds
     match &shape.shape_type {
         Type::Rect(_) | Type::Frame(_) => {
-            canvas.clip_rect(container, skia::ClipOp::Intersect, true);
+            if let Some(corners) = shape.shape_type.corners() {
+                let rrect: RRect = RRect::new_rect_radii(container, &corners);
+                canvas.clip_rrect(rrect, skia::ClipOp::Intersect, true);
+            } else {
+                canvas.clip_rect(container, skia::ClipOp::Intersect, true);
+            }
```

Comment:
We can take advantage of the pattern matching and have something like this to avoid yet another nested matching:

```rust
match &shape.shape_type {
  Type::Rect(Rect { corners: Some(corners), .. }) => { /* rounded rects */  },
  Type::Rect(Rect { corners: None, .. }) => { /* squared rect */ }
}
```